### PR TITLE
fix(components/headerbar): various improvements

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**
- Search bar doesn't follow the guidelines
- A separator can be displayed on the right of the last item
- HeaderBar doesn't use CSS module
- `getRenderers()` called in every methods
- Help cannot be displayed as an ActionSplitDropdown
- Dropdowns placed on the right of te nav can go beyond the screen

**What is the chosen solution to this problem?**
- Align search bar to guidelines
- Separators cannot be displayed on the right of the last item
- HeaderBar use a bit more CSS modules
- `getRenderers()` called a single time
- Help can be displayed as ActionSplitDropdown if these props has items
- `pullRight` property added the the dropdowns placed on the right of te nav

**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/HeaderBar/HeaderBar.scss [hunk 1]
```diff
@@ -88,10 +98,25 @@ $background-color-on-hover: shade($brand-primary, 30);
...
 	}
 
-	:global(.form-control) {
-		border-radius: 0;
-		color: $white;
-		font-style: italic;
-		height: 48px;
+	.tc-header-bar-search {
+		form {
+			height: 48px;
```
Comment (charlesnguyen) [inline packages/components/src/HeaderBar/HeaderBar.scss]: use $navbar-height?

File: output/components.eslint.txt [hunk 2]
```diff
@@ -19,6 +19,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   2:1  error  '@kadira/react-storybook-addon-info' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   7:2  error  Unexpected require()                                                                                      global-require
 
+/home/travis/build/Talend/ui/packages/components/src/HeaderBar/HeaderBar.component.js
+  135:11  error  Missing trailing comma  comma-dangle
```
Comment (charlesnguyen) [inline output/components.eslint.txt]: to be fixed

Comment (ncomont) [inline output/components.eslint.txt]: Already fixed, but the CI didn't update this report yet.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 3]
```diff
@@ -103,34 +127,44 @@ Search.propTypes.renderers = React.PropTypes.shape({
...
-		<li className="tc-header-bar-action">
-			<Components.Action
-				bsStyle="link"
-				icon="talend-question-circle"
-				label="Help"
-				tooltipPlacement="bottom"
-				{...props}
-			/>
+		<li className={theme['tc-header-bar-action']}>
+			{ props.items ? (
```
Comment (ncomont) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Done, thanks

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 4]
```diff
@@ -103,34 +127,44 @@ Search.propTypes.renderers = React.PropTypes.shape({
 });
 
 function Help({ renderers, ...props }) {
-	const Components = getRenderers(renderers);
+	const global = {
+		bsStyle: 'link',
+		icon: 'talend-question-circle',
+		label: 'Help',
+		tooltipPlacement: 'bottom',
+		...props,
```
Comment (jsomsanith) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Just to be sure, do we want to make the props possibly override one of the previous values, or are those more important ?

Comment (ncomont) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Previously, `...props` was placed after other props, so I think that `props` should be able to override fixed props.

**Pull request merged**